homeassistant/components/sun.py

```diff
@@ -124,6 +124,8 @@ def update_sun_position(self, utc_point_in_time):
     @callback
     def point_in_time_listener(self, now):
         """Run when the state of the sun has changed."""
+        if self.solar_elevation == STATE_UNKNOWN:
```

Comment:
Just drop the condition and always update the position. It's safe to update this here. Change it to use the time passed to the event:
```python
self.update_sun_position(now)
```